src/Microsoft.Tye.Hosting/PortAssigner.cs

```diff
@@ -30,24 +30,12 @@ public Task StartAsync(Application application)
                     continue;
                 }
 
-                static int GetNextPort()
-                {
-                    // Let the OS assign the next available port. Unless we cycle through all ports
-                    // on a test run, the OS will always increment the port number when making these calls.
-                    // This prevents races in parallel test runs where a test is already bound to
-                    // a given port, and a new test is able to bind to the same port due to port
-                    // reuse being enabled by default by the OS.
-                    using var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
-                    socket.Bind(new IPEndPoint(IPAddress.Loopback, 0));
-                    return ((IPEndPoint)socket.LocalEndPoint!).Port;
-                }
-
                 foreach (var binding in service.Description.Bindings)
                 {
                     // Auto assign a port
                     if (binding.Port == null)
                     {
-                        binding.Port = GetNextPort();
+                        binding.Port = NextPortFinder.Instance.GetNextPort();
```

Comment:
Why have NextPortFinder as an instance rather than just a static method?

Reply:
I followed the pattern from the `GitDetector` class but more than happy to make it static as it doesn't have any dependencies or logic in the constructor.